### PR TITLE
Edit existing report

### PR DESCRIPTION
**Pull request by idigra**

### CODE REVIEW

File: apps/vaccines-updates-client/src/components/main-form/index.tsx [hunk 1]
```diff
@@ -89,23 +92,35 @@ export const MainForm: FunctionComponent<{ className?: string }> = props => {
...
+
+            setReportIdToEdit(NewReport);
+            if (report.id) {
+                const index = availableReportsToEdit.reports.findIndex(r =>
+                    (r as { report: VaccinesReport }).report?.id?.pKey === report.id.pKey &&
+                    (r as { report: VaccinesReport }).report?.id?.internalId === report.id.internalId);
+                availableReportsToEdit.reports[index] = { report };
+            } else {
+                availableReportsToEdit.reports.push({ report });
+            }
```
Comment (YonatanKra) [inline apps/vaccines-updates-client/src/components/main-form/index.tsx]: This looks like a great function to extract from there. Maybe `createOrUpdateReport`?

Comment (idigra) [reply apps/vaccines-updates-client/src/components/main-form/index.tsx]: Done.

File: apps/vaccines-updates-client/src/components/main-form/parts/facebook-post/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    const [facebookPost, setFacebookPost] = useState<string>();
+    const [textAreaRows, setTextAreaRows] = useState<number>();
+
+    const getReports = useGetReports(/*returnHiddenReports=*/false);
+    const generatePost = useCallback(async () => {
+        // TODO: Error treatment
+        const reportsResponse = await getReports();
+        const reports = reportsResponse.reports.sort(function (a, b) {
+            let result = a.healthCareService.localeCompare(b.healthCareService);
+            // if (result === 0) result = a.serviceEndTime - b.serviceEndTime;
```
Comment (YonatanKra) [inline apps/vaccines-updates-client/src/components/main-form/parts/facebook-post/index.tsx:51]: What do you need this comment for?

**Review by YonatanKra — APPROVED** [review]
That's a huge PR, zero tests and I am not really versed in the app - so I approve just not to hold but maybe @gilm123 should take a look?

File: apps/vaccines-updates-client/src/components/main-form/parts/facebook-post/index.tsx (hunk 2 above)
Comment (idigra) [reply apps/vaccines-updates-client/src/components/main-form/parts/facebook-post/index.tsx:51]: Actually not sure. We didn't consider the sorting criterions, just wanted something to be consistent enough so users will not get confused. I removed the endTIme sorting when it became optional variable.
I'll leave this comment for now just to have a commit which exactly matches the deployed version, we can clean it or add it back in the future.

**Pull request merged**